### PR TITLE
Inomurko/exit started update

### PR DESCRIPTION
**Pull request by InoMurko**
- Implements ExitStarted that includes bytes of the origin transaction for the output.
- allows spending pending outputs (this should be either left as is or pending needs to turn into confirmed on block submission)
- refactored transaction decoding because the decoding function was doing multiple things
- release task for contracts is able to fetch database information on the fly from the env vars
- updated hackney to stable version
- turned on `show_sensitive_data_on_connection_error ` for time being, we will remove this later on
- ability to mock feefeed added

### CODE REVIEW

File: apps/engine/lib/engine/db/output/output_query.ex [hunk 1]
```diff
@@ -14,9 +14,12 @@ defmodule Engine.DB.Output.OutputQuery do
     Output |> usable() |> by_position(positions)
   end
 
+  # enabled transaction chaining would allow the state to also be pending!
+  # what needs to be done to avoid `o.state == ^:pending` is to
+  # mark all pending outputs to confirmed after their origin block gets mined (or submitted to contracts)
   defp usable(query) do
     from(o in query,
-      where: is_nil(o.spending_transaction_id) and o.state == ^:confirmed
+      where: is_nil(o.spending_transaction_id) and (o.state == ^:confirmed or o.state == ^:pending)
```
Comment (ayrat555) [inline apps/engine/lib/engine/db/output/output_query.ex:22]: outputs doesn't have indexes on `state` and `sepending_transaction_id`

**Review by ayrat555 — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -283,6 +284,8 @@ jobs:
     steps:
       - checkout
       - run: docker login -u $DOCKER_USER -p $DOCKER_PASS
+      - run: pip3 install docker-compose --upgrade
```
[the remark this answers is not in the document]
Comment (InoMurko) [reply .circleci/config.yml:287]:
```
ERROR: gcloud failed to load: /tmp/_MEIEHFgnz/libssl.so.1.1: version `OPENSSL_1_1_1' not found (required by /opt/circleci/.pyenv/versions/3.8.5/lib/python3.8/lib-dynload/_ssl.cpython-38-x86_64-linux-gnu.so)
    gcloud_main = _import_gcloud_main()
    import googlecloudsdk.gcloud_main
    from googlecloudsdk.api_lib.iamcredentials import util as iamcred_util
    from googlecloudsdk.api_lib.util import apis_internal
    from googlecloudsdk.core import properties
    from googlecloudsdk.core import config
    from oauth2client import client
    from oauth2client import transport
    import httplib2
    from httplib2.python3.httplib2 import *
    import ssl
    import _ssl             # if we can't import it, let the error propagate

This usually indicates corruption in your gcloud installation or problems with your Python interpreter.

Please verify that the following is the path to a working Python 2.7 or 3.5+ executable:
    /opt/circleci/.pyenv/versions/3.8.5/bin/python3

If it is not, please set the CLOUDSDK_PYTHON environment variable to point to a working Python 2.7 or 3.5+ executable.

If you are still experiencing problems, please reinstall the Cloud SDK using the instructions here:
    https://cloud.google.com/sdk/
Traceback (most recent call last):
  File "docker/credentials/store.py", line 80, in _execute
  File "subprocess.py", line 411, in check_output
  File "subprocess.py", line 512, in run
subprocess.CalledProcessError: Command '['/opt/google/google-cloud-sdk/bin/docker-credential-gcloud', 'get']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker/auth.py", line 264, in _resolve_authconfig_credstore
  File "docker/credentials/store.py", line 35, in get
  File "docker/credentials/store.py", line 93, in _execute
docker.credentials.errors.StoreError: Credentials store docker-credential-gcloud exited with "".

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "bin/docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 67, in main
  File "compose/cli/main.py", line 126, in perform_command
  File "compose/cli/main.py", line 1070, in up
  File "compose/cli/main.py", line 1066, in up
  File "compose/project.py", line 615, in up
  File "compose/service.py", line 356, in ensure_image_exists
  File "compose/service.py", line 1267, in pull
  File "compose/progress_stream.py", line 99, in get_digest_from_pull
  File "compose/service.py", line 1234, in _do_pull
  File "docker/api/image.py", line 396, in pull
  File "docker/auth.py", line 48, in get_config_header
  File "docker/auth.py", line 324, in resolve_authconfig
  File "docker/auth.py", line 235, in resolve_authconfig
  File "docker/auth.py", line 281, in _resolve_authconfig_credstore
docker.errors.DockerException: Credentials store error: StoreError('Credentials store docker-credential-gcloud exited with "".')
[15043] Failed to execute script docker-compose
Attaching to 
```

File: apps/engine/lib/engine/db/output/output_query.ex (hunk 1 above)
Comment (InoMurko) [reply apps/engine/lib/engine/db/output/output_query.ex:22]: are you saying this is making this query slow?

Comment (ayrat555) [reply apps/engine/lib/engine/db/output/output_query.ex:22]: yes

**Pull request merged**